### PR TITLE
[ci] Remove module collections from Alps config

### LINKED ISSUE #1938: remove module collection from cscs config
> Starting with CPE 21.04, PrgEnv module collections are deprecated

need to update https://github.com/eth-cscs/reframe/blob/master/config/cscs.py#L994

### PR DESCRIPTION
**Pull request by jgphpc**
Closes #1938

### CODE REVIEW

**Review by teojgo — CHANGES_REQUESTED** [review]
I think you can remove the `collection` completely.

File: config/cscs.py [hunk 1]
```diff
@@ -973,7 +973,7 @@
                 'eiger', 'pilatus'
             ],
             'modules': [
-                {'name': 'PrgEnv-aocc', 'collection': True}
+                {'name': 'PrgEnv-aocc', 'collection': False}
```
Comment (vkarak) [inline config/cscs.py:976]: There is no need for the extended syntax. Simply write it as `'modules': ['PrgEnv-aocc']` etc.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by teojgo — APPROVED** [review]
lgtm

**Comment by jgphpc** [comment]
When I run:
```
class XTest_new(rfm.CompileOnlyRegressionTest):
    m = parameter(['cray-mpich/8.1.4'])
```

I get:
```
  * Reason: key error: 'collection'
Traceback (most recent call last):
  File "/users/piccinal/reframe.git/reframe/frontend/executors/__init__.py", line 269, in _safe_call
    return fn(*args, **kwargs)
  File "/users/piccinal/reframe.git/reframe/core/hooks.py", line 36, in _fn
    func(obj, *args, **kwargs)
  File "/users/piccinal/reframe.git/reframe/core/pipeline.py", line 97, in _wrapped
    return fn(*args, **kwargs)
  File "/users/piccinal/reframe.git/reframe/core/pipeline.py", line 1250, in compile
    trap_errors=True
  File "/users/piccinal/reframe.git/reframe/core/schedulers/__init__.py", line 364, in prepare
    builder.write(runtime.emit_loadenv_commands(*environs))
  File "/users/piccinal/reframe.git/reframe/core/runtime.py", line 237, in emit_loadenv_commands
    _, commands = loadenv(*environs)
  File "/users/piccinal/reframe.git/reframe/core/runtime.py", line 224, in loadenv
    m, mod['collection'], mod['path'],
KeyError: 'collection'
```

**Comment by vkarak** [comment]
> When I run:
> 
> ```
> class XTest_new(rfm.CompileOnlyRegressionTest):
>     m = parameter(['cray-mpich/8.1.4'])
> ```
> 
> I get:
> 
> ```
>   * Reason: key error: 'collection'
> Traceback (most recent call last):
>   File "/users/piccinal/reframe.git/reframe/frontend/executors/__init__.py", line 269, in _safe_call
>     return fn(*args, **kwargs)
>   File "/users/piccinal/reframe.git/reframe/core/hooks.py", line 36, in _fn
>     func(obj, *args, **kwargs)
>   File "/users/piccinal/reframe.git/reframe/core/pipeline.py", line 97, in _wrapped
>     return fn(*args, **kwargs)
>   File "/users/piccinal/reframe.git/reframe/core/pipeline.py", line 1250, in compile
>     trap_errors=True
>   File "/users/piccinal/reframe.git/reframe/core/schedulers/__init__.py", line 364, in prepare
>     builder.write(runtime.emit_loadenv_commands(*environs))
>   File "/users/piccinal/reframe.git/reframe/core/runtime.py", line 237, in emit_loadenv_commands
>     _, commands = loadenv(*environs)
>   File "/users/piccinal/reframe.git/reframe/core/runtime.py", line 224, in loadenv
>     m, mod['collection'], mod['path'],
> KeyError: 'collection'
> ```
> 
> > {'name': 'PrgEnv-gnu', 'collection': False} works around it

I think this is a bug. Can you open one?

The `'collection': False` is not the "right" workaround. You don't need to use dictionaries. Just define the the modules in the classic way:

```python
'modules': ['PrgEnv-gnu']
```

**Comment by jgphpc** [comment]
https://github.com/eth-cscs/reframe/issues/1943

**Review by vkarak — APPROVED** [review]

**Pull request merged**